### PR TITLE
additional output for additem command

### PR DESCRIPTION
**Pull request by martindev446**
When removing items with the `additem` command:

If the user provides an amount too large of a negative amount(the target's inventory did not contain enough of the item), the command will now output the amount of items that could not be destroyed.

Example 1: 
`.additem 50045 3`
`.additem 50045 -4`
Annotation 2019-12-27 185659
Example 2:
`.additem 17 -1`
Annotation 2019-12-27 185612

**Changes proposed:**

-  Changed `Player::DestroyItemCount` to return the amount of destroyed items.
-  if the user provides a negative amount to `additem` that is too large, then this command will now print an additional message indicating the amount it failed to destroy.
- Added an additional TrinityString to the database for this purpose

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

**Issues addressed:**
Addresses number 1 in this issue: https://github.com/TrinityCore/TrinityCore/issues/22993

**Tests performed:**
Built on `Debug`, `RelWithDebInfo`, and `Release`
Tested `Debug`, `RelWithDebInfo`, and `Release`

**Known issues and TODO list:** (add/remove lines as needed)

- none that I can see.

**Notes**
- I'm a little worried about outputting two messages for one command. This might seem strange to the user. Maybe I should change the original output message instead?
  - Example format: `Removing itemID=#, amount=# from [Playername]: Success=# Failed=#`
- I hope I placed the new `LANG_REMOVEITEM_FAILURE` string in the correct place.

### CODE REVIEW

File: src/server/scripts/Commands/cs_misc.cpp [hunk 1]
```diff
@@ -1263,8 +1263,24 @@ class misc_commandscript : public CommandScript
...
+                handler->PSendSysMessage(LANG_REMOVEITEM, itemId, destroyedItemCount, handler->GetNameLink(playerTarget).c_str());
+
+                // check to see if we were unable to destroy all of the amount requested.
+                uint32 unableToDestroyItemCount = -count - destroyedItemCount;
+                if (unableToDestroyItemCount > 0) {
+                    // output message for the amount of items we couldn't destroy
+                    handler->PSendSysMessage(LANG_REMOVEITEM_FAILURE, itemId, unableToDestroyItemCount, handler->GetNameLink(playerTarget).c_str());
+                }
+            }
+            else if (destroyedItemCount == 0) {
```
Comment (jackpoz) [inline src/server/scripts/Commands/cs_misc.cpp]: The additional check == 0 is not needed here,  else is enough

Comment (martindev446) [reply src/server/scripts/Commands/cs_misc.cpp]: Thanks, fixed

File: src/server/scripts/Commands/cs_misc.cpp [hunk 2]
```diff
@@ -1263,8 +1263,24 @@ class misc_commandscript : public CommandScript
         // Subtract
         if (count < 0)
         {
-            playerTarget->DestroyItemCount(itemId, -count, true, false);
-            handler->PSendSysMessage(LANG_REMOVEITEM, itemId, -count, handler->GetNameLink(playerTarget).c_str());
+            uint32 destroyedItemCount = playerTarget->DestroyItemCount(itemId, -count, true, false);
+            
+            if (destroyedItemCount > 0) {
```
Comment (Shauren) [inline src/server/scripts/Commands/cs_misc.cpp]: Surprised noone mentioned this yet: codestyle, placement of opening brace { in all of your changes must be on new line

**Review by Shauren — CHANGES_REQUESTED** [review]

File: src/server/scripts/Commands/cs_misc.cpp (hunk 2 above)
Comment (martindev446) [reply src/server/scripts/Commands/cs_misc.cpp]: gotcha. i should have noticed that.

**Review by Shauren — APPROVED** [review]

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Pull request merged**